backend/models/course.model.ts

```diff
@@ -10,7 +10,7 @@ export interface Module {
 }
 
 export interface Course extends Document {
-  id: string;
+  _id: Types.ObjectId;
```

Comment:
Good addition, but does the Document interface not already have the `_id` attribute defined?